Source/JavaScriptCore/runtime/CallData.cpp

```diff
@@ -54,6 +54,10 @@ JSValue call(JSGlobalObject* globalObject, JSValue functionObject, const CallDat
 {
     VM& vm = globalObject->vm();
     ASSERT(callData.type == CallData::Type::JS || callData.type == CallData::Type::Native);
+#if ASSERT_ENABLED
+    for (size_t i = 0; i < args.size(); ++i)
+        ASSERT_WITH_MESSAGE(!args.at(i).isEmpty(), "Arg %zu is empty", i);
```

Comment:
If the value is a JSCell, it's worth also checking that the JSCell is valid memory so that it crashes earlier on.
```c++
if (args.at(i).isCell())
JSC::Integrity::auditCellFully(vm, args.at(i).asCell());
```